app/dashboard/views/domain_detail.py

```diff
@@ -122,23 +122,17 @@ def domain_detail_dns(custom_domain_id):
                 spf_errors = get_txt_record(custom_domain.domain)
 
         elif request.form.get("form-name") == "check-dkim":
-            dkim_record = get_cname_record("dkim._domainkey." + custom_domain.domain)
-            if dkim_record == dkim_cname:
+            dkim_errors = domain_validator.validate_dkim_records(custom_domain)
+            if len(dkim_errors) == 0:
                 flash("DKIM is setup correctly.", "success")
-                custom_domain.dkim_verified = True
```

Comment:
We should set `custom_domain.dkim_verified` here right?

Reply:
The `DomainValidator` does it. If it's already set it doesn't change it. If it's not set it will set it to the result of the test.